File: e2e-tests/playwright/java/src/test/java/net/gradido/e2e/tests/InvalidLoginTest.java
package net.gradido.e2e.tests;

import net.gradido.e2e.base.BaseTest;
import net.gradido.e2e.pages.LoginPage;
import net.gradido.e2e.components.Toasts;
import org.junit.jupiter.api.*;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class InvalidLoginTest extends BaseTest {
  private LoginPage loginPage;

  @BeforeEach
  void initPageObjects() {
      loginPage = new LoginPage(page);
  }

  @Test
  void invalidUserSeesError() {
      page.navigate("http://localhost:3000/login");
      loginPage.login("peter@lustig.de", "wrongpass");
      Toasts toast = new Toasts(page);
      toast.assertErrorToastVisible();
  }
}

File: e2e-tests/playwright/typescript/tests/InvalidLoginTest.ts
import { test, expect } from '@playwright/test'
import { LoginPage } from '../pages/LoginPage'
import { Toasts } from '../components/Toasts'

test('invalid login', async ({ page }) => {
  const loginPage = new LoginPage(page)
  await loginPage.goto()
  await loginPage.login('peter@lustig.de', 'wrongpass')
  
  const toast = new Toasts(page)
  await toast.assertErrorToastVisible()
  await page.waitForTimeout(50)
})